tests/flow/test_degree.py

```diff
@@ -0,0 +1,350 @@
+
+        result_set = self.graph.query(q).result_set
+
+        self.env.assertEqual(len(result_set), 2)
+
+        self.env.assertEqual(result_set[0][0], 'a0')
+        self.env.assertEqual(result_set[0][1], 2)
+
+        self.env.assertEqual(result_set[1][0], 'a1')
+        self.env.assertEqual(result_set[1][1], 2)
+
+    def test_labeled_incoming_dest(self):
+        # consider only incoming edges with destination nodes of a specific type
+
+        q = """CALL algo.degree({destination:'A', dir: 'incoming'}) YIELD node, degree
+               RETURN node.name, degree
+               ORDER BY node.name"""
+
+        result_set = self.graph.query(q).result_set
+
+        self.env.assertEqual(len(result_set), 3)
+
+        self.env.assertEqual(result_set[0][0], 'a0')
+        self.env.assertEqual(result_set[0][1], 1)
+
+        self.env.assertEqual(result_set[1][0], 'a1')
+        self.env.assertEqual(result_set[1][1], 1)
+
+        self.env.assertEqual(result_set[2][0], 'b')
+        self.env.assertEqual(result_set[2][1], 2)
+
+    def test_labeled_incoming_src_and_dest(self):
+        # consider only incoming edges with source and destination nodes of
+        # a specific type
+
+        q = """CALL algo.degree({source: 'B', destination:'A', dir: 'incoming'})
+               YIELD node, degree
+               RETURN node.name, degree
+               ORDER BY node.name"""
+
+        result_set = self.graph.query(q).result_set
+
+        self.env.assertEqual(len(result_set), 1)
+
+        self.env.assertEqual(result_set[0][0], 'b')
+        self.env.assertEqual(result_set[0][1], 2)
+
+    def test_none_existing_labels_relations(self):
+        # specify none existing labels and relationship types
+
+        queries = [
+            "CALL algo.degree({source: 'Z'})",
+            "CALL algo.degree({destination: 'Z'})",
+            "CALL algo.degree({source: 'Z', destination: 'Z'})",
+            "CALL algo.degree({relation: 'Z'})",
+            "CALL algo.degree({source: 'Z', relation: 'Z'})",
+            "CALL algo.degree({destination: 'Z', relation: 'Z'})",
+            "CALL algo.degree({source: 'Z', destination: 'Z', relation: 'Z'})",
+
+            "CALL algo.degree({source: 'Z', dir:'incoming'})",
+            "CALL algo.degree({destination: 'Z', dir:'incoming'})",
+            "CALL algo.degree({source: 'Z', destination: 'Z', dir: 'incoming'})",
+            "CALL algo.degree({relation: 'Z', dir: 'incoming'})",
+            "CALL algo.degree({source: 'Z', relation: 'Z', dir: 'incoming'})",
+            "CALL algo.degree({destination: 'Z', relation: 'Z', dir: 'incoming'})",
+            "CALL algo.degree({source: 'Z', destination: 'Z', relation: 'Z', dir: 'incoming'})"
+        ]
+
+        for q in queries:
+            result_set = self.graph.query(q).result_set
+            self.env.assertEqual(len(result_set), 0)
+
+    def test_tensors(self):
+        # try running algo.degree against a graph containing tensors
```

Comment:
why raise error and not calculate it as multi edge is just like single edge?

Reply:
Because that might cause confusion,
`CREATE (a)-[:R]->(b), (a)-[:R]->(b)`

When computing `degree` the user would expect `a's` out-degree to be 2, not 1.
To avoid this confusion I prefer to emit the error.